### PR TITLE
Fix map iteration

### PR DESCRIPTION
**Pull request by dwalluck**
Avoid (possibly multiple) `get()` calls when iterating over maps.

### CODE REVIEW

**Comment by rnc** [comment]
Good idea

**Comment by rnc** [comment]
While I'm fine with this and it looks correct to me, I guess _techinically_ using `map.forEach((k,v) -> { ....` would be slightly more efficient than a `for` and `Map.Entry` although _not_ applicable to every case - as break/continue/exceptions could cause problems with the lambda statements.

**Comment by dwalluck** [comment]
> While I'm fine with this and it looks correct to me, I guess _techinically_ using `map.forEach((k,v) -> { ....` would be slightly more efficient than a `for` and `Map.Entry` although _not_ applicable to every case - as break/continue/exceptions could cause problems with the lambda statements.

I don't see how. I believe the `Map.forEach()` implementation is basically equivalent to `Map.entrySet().forEach()`, so the performance should be nearly identical  to `for-each-entrySet()`, but with some slight overhead.

Also, in addition to the drawbacks of lambdas that you point out, lambdas can  sometimes get unwieldly when they're more than a few lines or nested.

In the current code, there was only a single instance of `Map.forEach()` that I recall, and I changed it to match the pattern of the others. It currently keeps this same pattern whether using `keySet()`, `values()`, or `entrySet()`.

File: core/src/main/java/org/commonjava/maven/ext/core/impl/SuffixManipulator.java [hunk 1]
```diff
@@ -136,17 +135,20 @@ private void processPlugins( Pattern suffixStripPattern, Project project,
         {
             if ( plugins != null )
             {
-                plugins.keySet().forEach( a -> {
```
Comment (dwalluck) [inline core/src/main/java/org/commonjava/maven/ext/core/impl/SuffixManipulator.java:139]: This was the only existing `.forEach()`.

**Review by rnc — APPROVED** [review]

**Pull request merged**